src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -833,6 +861,16 @@ class PersonAgent(
           } else {
             currentVehicle
           }
+        // todo rrp
+        currentVehicleForNextState.headOption.foreach { beamVehicleId =>
+          val vehicle = beamVehicles(beamVehicleId).vehicle
+          val refuelNeeded = vehicle.isRefuelNeeded(
+            beamScenario.beamConfig.beam.agentsim.agents.vehicles.rechargeRequiredThresholdInMeters,
+            beamScenario.beamConfig.beam.agentsim.agents.vehicles.noRechargeThresholdInMeters
+          )
+          if ((vehicle.isBEV || vehicle.isPHEV) && refuelNeeded)
```

Comment:
We also should look at the distance to destination.. if distance is short then let it follow default workflow, if destination is far then we plan enroute charging.
Maybe you can do with: 

val newRechargeRequiredThresholdInMeters = rechargeRequiredThresholdInMeters + (1-rechargeRequiredThresholdInMeters/distanceToDestinationInMeters)*distanceToDestinationInMeters

something similar should be done to noRechargeThresholdInMeters

Reply:
Got it. Correct me if I'm wrong, we need to use `distanceToDestinationInMeters` to come up with new thresholds (at run time) and use them to compute whether recharging it required or not. 

@haitamlaarabi, Could you please provide equation for `noRechargeThresholdInMeters` as well? I can't figure it out how do that.

Reply:
yes, we need to know if the battery will run low while driving towards destination. If it is not running low then we can leave workflow as it is. If it is running low then we need to EnRoute charge